src/health_manager.py

```diff
@@ -100,32 +101,45 @@ def start_monitor(self, run_thread: Thread):
             if not is_loading_black_roi:
                 # check health
                 health_percentage = self.get_health(img)
+                last_rejuv_drink = time.time() - self._last_rejuv
                 last_drink = time.time() - self._last_health
-                if health_percentage < self._config.char["take_health_potion"] and last_drink > 3.5:
-                    self._drink_poition(img, "health")
-                    self._last_health = time.time()
+
                 # give the chicken a 6 sec delay to give time for a healing pot and avoid endless loop of chicken
-                elif health_percentage < self._config.char["chicken"] and (time.time() - start) > 6:
+                if health_percentage < self._config.char["chicken"] and (time.time() - start) > 6:
                     Logger.warning(f"Trying to chicken, player HP {(health_percentage*100):.1f}%!")
                     self._do_chicken(img, run_thread)
                     break
+                elif health_percentage < self._config.char["take_rejuvenation_potion_health"] and last_rejuv_drink > 3:
+                    self._drink_potion("rejuvenation_potion",health_percentage)
+                    self._last_rejuv = time.time()
+                elif health_percentage < self._config.char["take_healing_potion"] and last_drink > 3.5 and last_rejuv_drink > 1.5:
+                    self._drink_potion("healing_potion",health_percentage)
+                    self._last_health = time.time()
                 # check mana
                 mana_percentage = self.get_mana(img)
                 last_drink = time.time() - self._last_mana
-                if mana_percentage < self._config.char["take_mana_potion"] and last_drink > 4:
-                    self._drink_poition(img, "mana")
+                last_rejuv_drink = time.time() - self._last_rejuv
+                if mana_percentage < self._config.char["take_rejuvenation_potion_mana"] and last_rejuv_drink > 4:
+                    self._drink_potion("rejuvenation_potion",mana_percentage)
+                    self._last_rejuv = time.time()
+                elif mana_percentage < self._config.char["take_mana_potion"] and last_drink > 4 and last_rejuv_drink > 2:
+                    self._drink_potion("mana_potion",mana_percentage)
                     self._last_mana = time.time()
                 # check merc
                 merc_alive, _ = self._template_finder.search("MERC", img, roi=self._config.ui_roi["merc_icon"])
                 if merc_alive:
                     merc_health_percentage = self.get_merc_health(img)
                     last_drink = time.time() - self._last_merc_healh
+                    last_merc_rejuv_drink = time.time() - self._last_merc_rejuv
                     if merc_health_percentage < self._config.char["merc_chicken"]:
```

Comment:
For merc we could leave chicken this way, cause there is a delay in giving rejuv and merc consuming it. While for own char, there is no delay